grove/grove_test.go

```diff
@@ -1,40 +1,123 @@
 package grove_test
 
 import (
+	"bytes"
 	"os"
 	"testing"
 
+	forest "git.sr.ht/~whereswaldon/forest-go"
 	"git.sr.ht/~whereswaldon/forest-go/grove"
+	"git.sr.ht/~whereswaldon/forest-go/testkeys"
 )
 
+// fakeFile implements the grove.File interface, but is entirely in-memory.
+// This helps speed testing.
+type fakeFile struct {
+	*bytes.Buffer
+	name string
+}
+
+var _ grove.File = &fakeFile{}
+
+func newFakeFile(name string, content []byte) *fakeFile {
+	return &fakeFile{
+		name:   name,
+		Buffer: bytes.NewBuffer(content),
+	}
+}
+
+func (f *fakeFile) Name() string {
+	return f.name
+}
+
+// needed to implement Close so that fakeFile is a io.ReadWriteCloser
+func (f *fakeFile) Close() error {
+	return nil
+}
+
+// fakeFS implements grove.FS, but is entirely in-memory.
 type fakeFS struct {
-	files map[string]*os.File
+	files map[string]*fakeFile
 }
 
 var _ grove.FS = fakeFS{}
 
 // Open opens the given path as an absolute path relative to the root
 // of the fakeFS
-func (r fakeFS) Open(path string) (*os.File, error) {
-	return r.files[path], nil
+func (r fakeFS) Open(path string) (grove.File, error) {
+	file, exists := r.files[path]
```

Comment:
I seem to be leaning all kinds of features with these reviews lol

Reply:
Yeah, this is really handy. If you use the single-variable form, it panics if the key is undefined within the map. This way ensures that it doesn't panic.